src/mlpack/tests/lsh_test.cpp

```diff
@@ -466,6 +466,145 @@ BOOST_AUTO_TEST_CASE(DeterministicNoMerge)
+  arma::mat rdata;
+  arma::mat qdata;
+  data::Load(trainSet, rdata, true);
+  data::Load(testSet, qdata, true);
+
+  // Run classic knn on reference set
+  KNN knn(rdata);
+  arma::Mat<size_t> groundTruth;
+  arma::mat groundDistances;
+  knn.Search(qdata, k, groundTruth, groundDistances);
+  
+  bool foundIncrease = 0;
+
+  for (size_t rep = 0; rep < repetitions; ++rep)
+  {
+    // train a model
+    LSHSearch<> multiprobeTest(
+        rdata,
+        numProj,
+        numTables,
+        hashWidth,
+        secondHashSize,
+        bucketSize);
+
+    double prevRecall = 0;
+    // search with varying number of probes
+    for (size_t p = 0; p < probeTrials; ++p)
+    {
+      arma::Mat<size_t> lshNeighbors;
+      arma::mat lshDistances; //move outside of loop for speed?
+      
+      multiprobeTest.Search(
+          qdata, 
+          k, 
+          lshNeighbors, 
+          lshDistances, 
+          0, 
+          numProbes[p]);
+
+      // compute recall
+      double recall = LSHSearch<>::ComputeRecall(lshNeighbors, groundTruth); //TODO: change to LSHSearch::ComputeRecall??
+      if (p > 0)
+      {
+        // more probes should at the very least not lower recall...
+        BOOST_REQUIRE_GE(recall, prevRecall);
+
+        // ... and should ideally increase it a bit
+        if (recall > prevRecall + epsilonIncrease)
+         foundIncrease = true;
+        prevRecall = recall;
+      }
+    }
+  }
+
+  BOOST_REQUIRE(foundIncrease);
+}
+
+/**
+ * Test: This is a deterministic test that verifies multiprobe LSH works
+ * correctly. To do this, we generate a query , q1, that will end up
+ * outside of the bucket of its nearest neighbors in C1. The neighbors will be
+ * found only by searching additional probing bins.
+ *
+ * We verify this actually is the case.
+ * 
+ * Note: This test doesn't work, because of the effect the random offsets of LSH
+ * have on the 1st level codes. We need to think of something better.
+ */
+/*
+BOOST_AUTO_TEST_CASE(MultiprobeDeterministicTest)
+{
+  math::RandomSeed(std::time(NULL));
+  // generate known deterministic clusters of points
+  const size_t N = 40;
+  arma::mat rdata;
+  GetPointset(N, rdata);
+
+  const int k = N/2;
+  const double hashWidth = 1;
+  const int secondHashSize = 99901;
+  const int bucketSize = 500;
+
+  // 1 table, projections on orthonormal plane
+  arma::cube projections(2, 2, 1);
+  projections(0, 0, 0) = 1;
+  projections(1, 0, 0) = 0;
+  projections(0, 1, 0) = 0;
+  projections(1, 1, 0) = 1;
+
+  // construct LSH object with given tables
+  LSHSearch<> lshTest(rdata, projections,
+                      hashWidth, secondHashSize, bucketSize);
+
+
+  // two points near the clusters but outside their bins. q1's lowest scoring
+  // perturbation vectors will map to C1 cluster's bins.
+  arma::mat q1;
+  q1 << 1.1 << arma::endr << 3.3; // vector [1.1, 3.3]
+
+  arma::Mat<size_t> neighbors;
+  arma::mat distances;
+
+  // Test that q1 simple search comes up empty
```

Comment:
A lot of the comments in this test don't have capitalization or periods or aren't complete sentences.